tools/tf2openapi/Makefile

```diff
@@ -0,0 +1,36 @@
+VENDOR_DIR := $(GOPATH)/src/github.com/kubeflow/kfserving/vendor
+TF_PROTO_SRC := $(VENDOR_DIR)/github.com/tensorflow/tensorflow
+TF_PROTO_OUT := tf
+
+# Generate code
+generate:
+ifndef GOPATH
+	$(error GOPATH not defined, please define GOPATH. Run "go help gopath" to learn more about GOPATH)
+endif
+
+ifeq (,$(findstring $(GOPATH)/bin, $(PATH)))
+	$(error GOPATH bin not in path, please add it to PATH)
+endif
+	protoc -I $(TF_PROTO_SRC) --go_out=$(VENDOR_DIR) $(TF_PROTO_SRC)/tensorflow/core/lib/core/error_codes.proto
```

Comment:
Why output these to VENDORDIR? Shouldn't we just output them into TF_PROTO_OUT?

Reply:
`go_package` is set to `github.com/tensorflow/tensorflow...` in the proto sources, so `protoc` expects to find `github.com/tensorflow/tensorflow...` in the output dir. If we output to TF_PROTO_OUT, `protoc` will create the corresponding directories for each level of `github.com/tensorflow/tensorflow...` within TF_PROTO_OUT. In that case, I think it would be more confusing to move the protos to TF_PROTO_OUT and clean up the directories than to output to VENDOR_DIR.

Reply:
Discussde offline